docs/options.md
Options
=======

Security Monkey's behavior can be adjusted with options passed using a configuration file or directly using the command line. Some parameters are only available in the configuration file.

If an option is not passed, Security Monkey will use the default value from the file security\_monkey/default-config.py.

You also have the option of providing environment aware configurations through the use of the SECURITY\_MONKEY\_SETTINGS environmental variable.

Any variables set via this variable will override the default values specified in default-config.py

Config File
-----------

### LOG\_LEVEL

Standard python logging levels (ERROR, WARNING, DEBUG) depending on how much output you would like to see in your logs.

### LOG\_FILE

If set, specifies a file to which Security Monkey will write logs. If unset, Security Monkey will log to stderr.

### LOG\_CFG

Can be used instead of LOG\_LEVEL and LOG\_FILE. Should be set to a PEP-0391 compatible logging configuration. Example:

    LOG_CFG = {
        'version': 1,
        'disable_existing_loggers': False,
        'formatters': {
            'standard': {
                'format': '%(asctime)s %(levelname)s: %(message)s '
                    '[in %(pathname)s:%(lineno)d]'
            }
        },
        'handlers': {
            'file': {
                'class': 'logging.handlers.RotatingFileHandler',
                'level': 'DEBUG',
                'formatter': 'standard',
                'filename': '/var/log/security_monkey/securitymonkey.log',
                'maxBytes': 10485760,
                'backupCount': 100,
                'encoding': 'utf8'
            },
            'console': {
                'class': 'logging.StreamHandler',
                'level': 'DEBUG',
                'formatter': 'standard',
                'stream': 'ext://sys.stdout'
            }
        },
        'loggers': {
            'security_monkey': {
                'handlers': ['file', 'console'],
                'level': 'DEBUG'
            },
            'apscheduler': {
                'handlers': ['file', 'console'],
                'level': 'INFO'
            }
        }
    }

### R53

Specify if you want Security Monkey to create a DNS entry for itself and what DNS name you would like

### FQDN

This is used for various redirection magic that to get the Security Monkey UI working nice with the API

### SQLALCHEMY\_POOL\_SIZE & SQLALCHEMY\_MAX\_OVERFLOW

Because of the parallel nature of Security Monkey we have to have the ability to tweak the number of concurrent connections we can make. The default values should be sufficient for \<= 20 accounts. This may need to be increased if you are dealing with a greater number of accounts.

### API\_PORT

Needed for CORS whitelisting -- this should match the port you have told Security Monkey to listen on. If you are using nginx it should match the port that nginx is listening on for the /api endpoint.

### WEB\_PORT

Needed for CORS whitelisting -- this should match the port you have configured nginx to listen on for static content.

### WEB\_PATH

### FQDN

To perform redirection security monkey needs to know the FQDN you intend to use. IF R53 is enabled this FQDN will be automatically added to Route53 when Security Monkey starts, assuming the SecurityMonkeyInstanceProfile has permission to do so.

### SQLACHEMY\_DATABASE\_URI

If you have ever used sqlalchemy before this is the standard connection string used. Security Monkey uses a postgres database and the connection string would look something like:

    SQLALCHEMY_DATABASE_URI = 'postgressql://<user>:<password>@<hostname>:5432/SecurityMonkey'

### SECRET_KEY

This `SECRET_KEY` is essential to ensure the sessions generated by Flask cannot be guessed. You must generate a RANDOM SECRET\_KEY for this value.

An example of how you might generate a random string:

    >>> import random
    >>> secret_key = ''.join(random.choice(string.ascii_uppercase) for x in range(6))
    >>> secret_key = secret_key + ''.join(random.choice("~!@#$%^&*()_+") for x in range(6))
    >>> secret_key = secret_key + ''.join(random.choice(string.ascii_lowercase) for x in range(6))
    >>> secret_key = secret_key + ''.join(random.choice(string.digits) for x in range(6))

### SECURITY\_PASSWORD\_SALT

For many of the same reasons we want want a random SECRET\_KEY we want to ensure our password salt is random. see: [Salt](http://en.wikipedia.org/wiki/Salt_(cryptography))

You can use the same method used to generate the SECRET\_KEY to generate the SECURITY\_PASSWORD\_SALT

### Additional Options

As Security Monkey uses Flask-Security for authentication see .. \_Flask-Security: <https://pythonhosted.org/Flask-Security/configuration.html> for additional configuration options.

Command line
------------

### --host and --port

The host and port on which to listen for incoming request. Usually 127.0.0.1 and 8000 to listen locally or 0.0.0.0 and 80 to listen from the outside.

Default: 127.0.0.1 and 8000

Setting file : HOST and PORT

### --version and --help

Display the help or the version of 0bin.

Default: None

Configuration file equivalent: None


docs/options.rst
============
Options
============

Security Monkey's behavior can be adjusted with options passed using a configuration
file or directly using the command line. Some parameters are only available
in the configuration file.

If an option is not passed, Security Monkey will use the default value from the file
security_monkey/default-config.py.

You also have the option of providing environment aware configurations through the use
of the SECURITY_MONKEY_SETTINGS environmental variable.

Any variables set via this variable will override the default values specified in default-config.py


Config File
===========

LOG_LEVEL
---------

Standard python logging levels (ERROR, WARNING, DEBUG) depending on how much output you would like to see in your logs.

LOG_FILE
--------

If set, specifies a file to which Security Monkey will write logs. If unset, Security Monkey will log to stderr.

LOG_CFG
-------
Can be used instead of LOG_LEVEL and LOG_FILE.  Should be set to a PEP-0391 compatible logging configuration.  Example::

        LOG_CFG = {
            'version': 1,
            'disable_existing_loggers': False,
            'formatters': {
                'standard': {
                    'format': '%(asctime)s %(levelname)s: %(message)s '
                        '[in %(pathname)s:%(lineno)d]'
                }
            },
            'handlers': {
                'file': {
                    'class': 'logging.handlers.RotatingFileHandler',
                    'level': 'DEBUG',
                    'formatter': 'standard',
                    'filename': '/var/log/security_monkey/securitymonkey.log',
                    'maxBytes': 10485760,
                    'backupCount': 100,
                    'encoding': 'utf8'
                },
                'console': {
                    'class': 'logging.StreamHandler',
                    'level': 'DEBUG',
                    'formatter': 'standard',
                    'stream': 'ext://sys.stdout'
                }
            },
            'loggers': {
                'security_monkey': {
                    'handlers': ['file', 'console'],
                    'level': 'DEBUG'
                },
                'apscheduler': {
                    'handlers': ['file', 'console'],
                    'level': 'INFO'
                }
            }
        }

R53
---

Specify if you want Security Monkey to create a DNS entry for itself and what DNS name you would like

FQDN
----

This is used for various redirection magic that to get the Security Monkey UI working nice with the API


SQLALCHEMY_DATABASE_URI
-----------------------

Specify where you would like Security Monkey to store it's results

SQLALCHEMY_POOL_SIZE & SQLALCHEMY_MAX_OVERFLOW
----------------------------------------------

Because of the parallel nature of Security Monkey we have to have the ability to tweak the number of concurrent connections we can make. The default values should be sufficient for <= 20 accounts. This may need to be increased if you are dealing with a greater number of accounts.

API_PORT
-------- 

Needed for CORS whitelisting -- this should match the port you have told Security Monkey to listen on. If you are using nginx it should match the port that nginx is listening on for the /api endpoint.

WEB_PORT
--------

Needed for CORS whitelisting -- this should match the port you have configured nginx to listen on for static content.

WEB_PATH
--------


Additional Options
------------------

As Security Monkey uses Flask-Security for authentication see .. _Flask-Security: https://pythonhosted.org/Flask-Security/configuration.html for additional configuration options.


Command line
==================

--host and --port
-------------------

The host and port on which to listen for incoming request. Usually 127.0.0.1
and 8000 to listen locally or 0.0.0.0 and 80 to listen from the outside.

Default: 127.0.0.1 and 8000

Setting file : HOST and PORT

--version and --help
--------------------

Display the help or the version of 0bin.

Default: None

Configuration file equivalent: None



docs/changelog.md
Changelog
=========

v0.9.2 (2017-05-24)
----------------------------------------

- PR #695 - @mikegrima - Fixing jinja import bug affecting change emails.
- PR #692 - @LukeKennedy - Reduce number of API calls in Managed Policy watcher.
- PR #694 - @supertom - GCP Documentation Updates
- PR #701 - @supertom - Update GCP ServiceAccount Name to use email instead of DisplayName.
- PR #702 - @rodriguezsergio - Update KMS Auditor. Don't create issue when Effect is Deny for a wildcard principal.
- PR #697 - @mcpeak - Pylint fixes and TravisCI pylint enforcement.
- PR #706 - @monkeysecurity Fix bug where batched watchers did not send change alert emails.
- PR #708 - @redixin - Fix bug in docker config where `SECURITY_MONKEY_POSTGRES_PORT` would not work if passed as a string.
- PR #714 - @monkeysecurity - Fix bug where change emails from batched watchers had incorrect color in the JSON diff.
- PR #713 - @monkeysecurity - Fix path to favicon from flask-security jinja templates.
- PR #709 - @crruthe - Exempt SSO API from CSRF protection.
- PR #719 - @monkeysecurity - New simplified watcher format for CloudAux Technologies.
- PR #726 - @monkeysecurity, @willbengtson - Add new SAMLProvider watcher.
- PR #730 - @monkeysecurity - Fix bug where ephemerals were not respected for CloudAuxWatcher subclasses.
- PR #727 - @supertom - Fix bug where duplicate GCP names would violate DB's unique constraint. Names now contain project ID.
- PR #728 - @supertom - Basic Auditor Tests for GCP.
- @monkeysecurity - Updated link to Ubuntu's SSL documentation.
- @monkeysecurity - Bumped version of Cryptography dependency.
- PEP8 updates.

Important Notes:
- Additional Permissions Required:
    - "elasticloadbalancing:describelisteners",
    - "elasticloadbalancing:describerules",
    - "elasticloadbalancing:describesslpolicies",
    - "elasticloadbalancing:describetags",
    - "elasticloadbalancing:describetargetgroups",
    - "elasticloadbalancing:describetargetgroupattributes",
    - "elasticloadbalancing:describetargethealth",
    - "iam:listsamlproviders",
- New Watcher: ALB (elbv2)
- ELB (v1) Watcher re-written with boto3 in CloudAux.  Now respects the config value `SECURITYGROUP_INSTANCE_DETAIL` when determining whether to add the instance id's to the ELB definition.
 
Contributors:
- @LukeKennedy
- @rodriguezsergio
- @redixin
- @crruthe
- @supertom
- @mcpeak
- @mikegrima
- @monkeysecurity

v0.9.1 (2017-04-20)
----------------------------------------

- PR #666 - @redixin - Use find_packages in setup.py to include nested packages.
- PR #667 - @monkeysecurity - Explicitly adding `urllib3[secure]` to setup.py (REVERTED in #683)
- PR #668 - @monkeysecurity - IPv6 support in security groups.
- PR #669 - @monkeysecurity - Updating the security group auditor to treat `::/0` the same as `0.0.0.0/0`
- PR #671 - @monkeysecurity - Enhancing PolicyDiff to be able to handle non-ascii strings.
- PR #673 - @monkeysecurity - Fixing path to `aws_accounts.json`. (Broken my moving `manage.py`)
- PR #675 - @monkeysecurity - Adding `package_data` and `data_files` sections to setup.py.
- PR #677 - @willbengtson - Fixing the security trackable information.
- PR #682 - @monkeysecurity - Updating packaged supervisor config to provide full path to `monkey`
- PR #681 - @AlexCline - Add reference_policies for TLS transitional ELB security policies
- PR #684 - @monkeysecurity - Disabling DB migration `b8ccf5b8089b`. Was freezing some `db upgrades`
- PR #683 - @monkeysecurity - Reverted #667.  Added `pip install --upgrade urllib3[secure]` to `quickstart` and `Dockerfile`.
- PR #685 - @monkeysecurity - Running `docker-compose build` in Travis-CI.
- PR #688 - @mcpeak - Add Bandit gate to Security Monkey.
- PR #687 - @mikegrima - Fix for issue #680. (Unable to edit account names)
- PR #689 - @mikegrima - Enhancements to Travis-CI: parallelized the workloads. (docker/python/dart in parallel)

Important Notes:
 - This is a hotfix release to correct a number of installation difficulties reported since `0.9.0`.

Contributors:
- @redixin
- @AlexCline
- @willbengtson
- @mcpeak
- @mikegrima
- @monkeysecurity

v0.9.0 (2017-04-13)
----------------------------------------

- PR #500 - @monkeysecurity - Updating ARN.py to look for StringEqualsIgnoreCase in policy condition blocks
- PR #511 - @kalpatel01 - Fix KMSAuditor exceptions
- PR #510 - @kalpatel01 - Add additional JIRA configurations
- PR #504 - @redixin - Plugins support
- PR #515 - @badraufran - Add ability to press enter to search in search bar component
- PR #514 - @badraufran - Update dev_setup_osx.rst to get it up-to-date
- PR #513 / #545- @mikegrima - Fix for S3 watcher errors.
- PR #516  - @badraufran - Remove broken packages link
- PR #518 - @badraufran  - Update `dev_setup_osx` (Remove sudo)
- PR #519 - @selmanj - Minor reformatting/style changes to Docker docs
- PR #512 / #521 - @kalpatel01 - Organize tests into directories
- PR #524 - @kalpatel01 - Remove DB mock class
- PR #522 - @kalpatel01 - Optimize SQL for account delete 
- PR #525 - @kalpatel01 - Handle known kms boto exceptions
- PR #529 - @mariusgrigaitis - Usage of `GOOGLE_HOSTED_DOMAIN` in sample configs
- PR #532 - @kalpatel01 - Add sorting to account tables (UI)
- PR #538 - @cu12 - Add more Docker envvars
- PR #536 / #540 - @supertom - Add account type field to item, item details and search bar.
- PR #534 / #541 - @kalpatel01 - Add bulk enable and disable account service
- PR #546 - @supertom - GCP: fixed accounttypes typo.
- PR #547 - @monkeysecurity - Delete deprecated Account fields
- PR #528 - @kalpatel01 - Fix reaudit issue for watchers in different intervals
- PR #553 - @mikegrima - Fixed bugs in the ES watcher
- PR #535 / #552 - @kalpatel01 - Add support for overriding audit scores
- PR #560 / #587 - @mikegrima - Bump CloudAux version
- PR #533 / #559 - @kalpatel01 - Add Watcher configuration
- PR #562 - @monkeysecurity - Re-adding reporter timing information to the logs.

env-config/config.py
#     Copyright 2014 Netflix, Inc.
#
#     Licensed under the Apache License, Version 2.0 (the "License");
#     you may not use this file except in compliance with the License.
#     You may obtain a copy of the License at
#
#         http://www.apache.org/licenses/LICENSE-2.0
#
#     Unless required by applicable law or agreed to in writing, software
#     distributed under the License is distributed on an "AS IS" BASIS,
#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#     See the License for the specific language governing permissions and
#     limitations under the License.
# Insert any config items here.
# This will be fed into Flask/SQLAlchemy inside security_monkey/__init__.py

LOG_CFG = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'standard': {
            'format': '%(asctime)s %(levelname)s: %(message)s '
                '[in %(pathname)s:%(lineno)d]'
        }
    },
    'handlers': {
        'file': {
            # 'class': 'logging.handlers.RotatingFileHandler',
            'class': 'logging.handlers.GroupWriteRotatingFileHandler',
            'level': 'DEBUG',
            'formatter': 'standard',
            'filename': '/var/log/security_monkey/securitymonkey.log',
            'maxBytes': 10485760,
            'backupCount': 100,
            'encoding': 'utf8'
        },
        'console': {
            'class': 'logging.StreamHandler',
            'level': 'DEBUG',
            'formatter': 'standard',
            'stream': 'ext://sys.stdout'
        }
    },
    'loggers': {
        'security_monkey': {
            'handlers': ['file', 'console'],
            'level': 'WARN'
        },
        'apscheduler': {
            'handlers': ['file', 'console'],
            'level': 'INFO'
        }
    }
}

SQLALCHEMY_DATABASE_URI = 'postgresql://securitymonkeyuser:securitymonkeypassword@localhost:5432/secmonkey'

SQLALCHEMY_POOL_SIZE = 50
SQLALCHEMY_MAX_OVERFLOW = 15
ENVIRONMENT = 'ec2'
USE_ROUTE53 = False
FQDN = 'ec2-XX-XXX-XXX-XXX.compute-1.amazonaws.com'
API_PORT = '5000'
WEB_PORT = '443'
WEB_PATH = '/static/ui.html'
FRONTED_BY_NGINX = True
NGINX_PORT = '443'
BASE_URL = 'https://{}/'.format(FQDN)

SECRET_KEY = '<INSERT_RANDOM_STRING_HERE>'

MAIL_DEFAULT_SENDER = 'securitymonkey@example.com'
SECURITY_REGISTERABLE = False
SECURITY_CONFIRMABLE = False
SECURITY_RECOVERABLE = False
SECURITY_PASSWORD_HASH = 'bcrypt'
SECURITY_PASSWORD_SALT = '<INSERT_RANDOM_STRING_HERE>'
SECURITY_TRACKABLE = True

SECURITY_POST_LOGIN_VIEW = BASE_URL
SECURITY_POST_REGISTER_VIEW = BASE_URL
SECURITY_POST_CONFIRM_VIEW = BASE_URL
SECURITY_POST_RESET_VIEW = BASE_URL
SECURITY_POST_CHANGE_VIEW = BASE_URL

# This address gets all change notifications (i.e. 'securityteam@example.com')
SECURITY_TEAM_EMAIL = []

# These are only required if using SMTP instead of SES
EMAILS_USE_SMTP = False     # Otherwise, Use SES
SES_REGION = 'cn-north-1'
MAIL_SERVER = 'smtp.example.com'
MAIL_PORT = 465
MAIL_USE_SSL = True
MAIL_USERNAME = 'username'
MAIL_PASSWORD = 'password'

WTF_CSRF_ENABLED = True
WTF_CSRF_SSL_STRICT = True # Checks Referer Header. Set to False for API access.
WTF_CSRF_METHODS = ['DELETE', 'POST', 'PUT', 'PATCH']

# "NONE", "SUMMARY", or "FULL"
SECURITYGROUP_INSTANCE_DETAIL = 'FULL'

# Threads used by the scheduler.
# You will likely need at least one core thread for every account being monitored.
CORE_THREADS = 25
MAX_THREADS = 30

# SSO SETTINGS:
ACTIVE_PROVIDERS = []  # "ping", "google" or "onelogin"

PING_NAME = ''  # Use to override the Ping name in the UI.
PING_REDIRECT_URI = "{BASE}api/1/auth/ping".format(BASE=BASE_URL)
PING_CLIENT_ID = ''  # Provided by your administrator
PING_AUTH_ENDPOINT = ''  # Often something ending in authorization.oauth2
PING_ACCESS_TOKEN_URL = ''  # Often something ending in token.oauth2
PING_USER_API_URL = ''  # Often something ending in idp/userinfo.openid
PING_JWKS_URL = ''  # Often something ending in JWKS
PING_SECRET = ''  # Provided by your administrator
PING_DEFAULT_ROLE = 'View'

GOOGLE_CLIENT_ID = ''
GOOGLE_AUTH_ENDPOINT = ''
GOOGLE_SECRET = ''
# GOOGLE_HOSTED_DOMAIN = 'example.com' # Verify that token issued by comes from domain
GOOGLE_DEFAULT_ROLE = 'View'

ONELOGIN_APP_ID = '<APP_ID>'  # OneLogin App ID provider by your administrator
ONELOGIN_EMAIL_FIELD = 'User.email'  # SAML attribute used to provide email address
ONELOGIN_DEFAULT_ROLE = 'View'  # Default RBAC when user doesn't already exist
ONELOGIN_HTTPS = True  # If using HTTPS strict mode will check the requests are HTTPS
ONELOGIN_SETTINGS = {
    # If strict is True, then the Python Toolkit will reject unsigned
    # or unencrypted messages if it expects them to be signed or encrypted.
    # Also it will reject the messages if the SAML standard is not strictly
    # followed. Destination, NameId, Conditions ... are validated too.
    "strict": True,

    # Enable debug mode (outputs errors).
    "debug": True,

    # Service Provider Data that we are deploying.
    "sp": {
        # Identifier of the SP entity  (must be a URI)
        "entityId": "{BASE}metadata/".format(BASE=BASE_URL),
        # Specifies info about where and how the <AuthnResponse> message MUST be
        # returned to the requester, in this case our SP.
        "assertionConsumerService": {
            # URL Location where the <Response> from the IdP will be returned
            "url": "{BASE}api/1/auth/onelogin?acs".format(BASE=BASE_URL),
            # SAML protocol binding to be used when returning the <Response>
            # message. OneLogin Toolkit supports this endpoint for the
            # HTTP-POST binding only.
            "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST"
        },
        # If you need to specify requested attributes, set a
        # attributeConsumingService. nameFormat, attributeValue and
        # friendlyName can be omitted
        #"attributeConsumingService": {
        #        "ServiceName": "SP test",
        #        "serviceDescription": "Test Service",
        #        "requestedAttributes": [
        #            {
        #                "name": "",
        #                "isRequired": False,
        #                "nameFormat": "",
        #                "friendlyName": "",
        #                "attributeValue": ""
        #            }
        #        ]
        #},
        # Specifies info about where and how the <Logout Response> message MUST be
        # returned to the requester, in this case our SP.
        "singleLogoutService": {
            # URL Location where the <Response> from the IdP will be returned
            "url": "{BASE}api/1/auth/onelogin?sls".format(BASE=BASE_URL),
            # SAML protocol binding to be used when returning the <Response>
            # message. OneLogin Toolkit supports the HTTP-Redirect binding
            # only for this endpoint.
            "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
        },
        # Specifies the constraints on the name identifier to be used to
        # represent the requested subject.
        # Take a look on src/onelogin/saml2/constants.py to see the NameIdFormat that are supported.
        "NameIDFormat": "urn:oasis:names:tc:SAML:1.1:nameid-format:unspecified",
        # Usually x509cert and privateKey of the SP are provided by files placed at
        # the certs folder. But we can also provide them with the following parameters
        "x509cert": "",
        "privateKey": ""
    },

    # Identity Provider Data that we want connected with our SP.
    "idp": {
        # Identifier of the IdP entity  (must be a URI)
        "entityId": "https://app.onelogin.com/saml/metadata/{APP_ID}".format(APP_ID=ONELOGIN_APP_ID),
        # SSO endpoint info of the IdP. (Authentication Request protocol)
        "singleSignOnService": {
            # URL Target of the IdP where the Authentication Request Message
            # will be sent.
            "url": "https://app.onelogin.com/trust/saml2/http-post/sso/{APP_ID}".format(APP_ID=ONELOGIN_APP_ID),
            # SAML protocol binding to be used when returning the <Response>
            # message. OneLogin Toolkit supports the HTTP-Redirect binding
            # only for this endpoint.
            "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
        },
        # SLO endpoint info of the IdP.
        "singleLogoutService": {
            # URL Location of the IdP where SLO Request will be sent.
            "url": "https://app.onelogin.com/trust/saml2/http-redirect/slo/{APP_ID}".format(APP_ID=ONELOGIN_APP_ID),
            # SAML protocol binding to be used when returning the <Response>
            # message. OneLogin Toolkit supports the HTTP-Redirect binding
            # only for this endpoint.
            "binding": "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-Redirect"
        },
        # Public x509 certificate of the IdP
        "x509cert": "<ONELOGIN_APP_CERT>"
    }
}

from datetime import timedelta
PERMANENT_SESSION_LIFETIME=timedelta(minutes=60)
SESSION_REFRESH_EACH_REQUEST=True
SESSION_COOKIE_SECURE=True
SESSION_COOKIE_HTTPONLY=True
PREFERRED_URL_SCHEME='https'

REMEMBER_COOKIE_DURATION=timedelta(minutes=60)  # Can make longer if you want remember_me to be useful.
REMEMBER_COOKIE_SECURE=True
REMEMBER_COOKIE_HTTPONLY=True

# Apscheduler Configurations
# Length of time, in seconds, before a scheduled job is cancelled due to thread contention or other issues
MISFIRE_GRACE_TIME=30
# Delay, in seconds, until reporter starts
REPORTER_START_DELAY=10

# JIRA Settings
# Verify JIRA SSL certs - useful for testing on JIRA sandbox server
JIRA_SSL_VERIFY = True
JIRA_OPEN = 'Open' # Opened ticket JIRA transition name (e.g. 'Open', 'To Do')
JIRA_CLOSED = 'Closed' # Closed ticket JIRA transition name (e.g. 'Closed', 'Done')
